docs/tutorials/basic/ndarray_indexing.md

```diff
@@ -0,0 +1,375 @@
+
+# NDArray Indexing - Array indexing features
+
+MXNet's advanced indexing features are modeled after [NumPy's implementation and documentation](https://docs.scipy.org/doc/numpy-1.13.0/reference/arrays.indexing.html#combining-advanced-and-basic-indexing). You will see direct adaptations of many NumPy indexing features, and these are close, if not identical.
```

Comment:
Should we also emphasize that some examples and explanations are borrowed from NumPy's indexing tutorial directly for illustrating the indexing feature of MXNet?